Fail to build with debian 12 packaging

```
 aarch64-linux-gnu-g++ -g -O2 -ffile-prefix-map=/workspace/build=. -fstack-protector-strong -Wformat -Werror=format-security -MT test/Slice/parser/build/aarch64-linux-gnu/obj/pic/CircularB.o -MMD -MP -MF  test/Slice/parser/build/aarch64-linux-gnu/obj/pic/CircularB.Td -fvisibility=hidden -Wall -Wextra -Wredundant-decls -Wshadow -Wdeprecated -Werror -pthread -g -std=c++17 -Itest/Slice/parser/generated -Itest/include -I../cpp/include -I../cpp/include/generated -fPIC -Wdate-time -D_FORTIFY_SOURCE=2 -c test/Slice/parser/generated/CircularB.cpp -o test/Slice/parser/build/aarch64-linux-gnu/obj/pic/CircularB.o
test/Slice/escape/Client.cpp: In function ‘void testtypes(const Ice::CommunicatorPtr&)’:
test/Slice/escape/Client.cpp:88:20: error: array subscript -5 is outside array bounds of ‘int (* [1152921504606846975])(...)’ [-Werror=array-bounds]
   88 |     g->escaped_case(0, d2);
      |     ~~~~~~~~~~~~~~~^~~~~~~
```

--- a/cpp/test/Slice/escape/Client.cpp
+++ b/cpp/test/Slice/escape/Client.cpp
@@ -81,13 +81,13 @@ testtypes(const Ice::CommunicatorPtr& communicator)
 
 // Work-around for:
 // error: array subscript -6 is outside array bounds of ‘int (* [1152921504606846975])(...)’ [-Werror=array-bounds]
-#if defined(NDEBUG) && defined(__GNUC__)
+#if defined(__GNUC__)
 #    pragma GCC diagnostic push
 #    pragma GCC diagnostic ignored "-Warray-bounds"
 #endif
     g->escaped_case(0, d2);
     g->escaped_explicit();
-#if defined(NDEBUG) && defined(__GNUC__)
+#if defined(__GNUC__)
 #    pragma GCC diagnostic pop
 #endif
     escaped_and::escaped_doPtr g1 = std::make_shared<doI>();

--- a/packaging/dpkg/debian/rules
+++ b/packaging/dpkg/debian/rules
@@ -54,7 +54,7 @@ ifeq (,$(filter no-python312,$(DEB_BUILD_PROFILES)))
 	$(MAKE) $(MAKEOPTS) OPTIMIZE=$(OPTIMIZE) PYTHON=python3 LANGUAGES="python" srcs
 endif
 ifeq (,$(filter nocheck,$(DEB_BUILD_OPTIONS)))
-	$(MAKE) $(MAKEOPTS) OPTIMIZE=no LANGUAGES=cpp CONFIGS="shared" tests
+	$(MAKE) $(MAKEOPTS) OPTIMIZE=$(OPTIMIZE) LANGUAGES=cpp CONFIGS="shared" tests
 endif
 	for v in $(PHP_VERSIONS); do \
 	  cp -a php php-$${v}; \

